cmd/grr/workflow.go

```diff
@@ -287,10 +288,22 @@ func initialiseCmd(cmd *cli.Command, opts *grizzly.Opts) *cli.Command {
 
 	cmd.Flags().StringSliceVarP(&opts.Targets, "target", "t", nil, "resources to target")
 	cmd.Flags().StringSliceVarP(&opts.JsonnetPaths, "jpath", "J", getDefaultJsonnetFolders(), "Specify an additional library search dir (right-most wins)")
+	cmd.Flags().StringVarP(&opts.OutputFormat, "output", "o", "", "Output format")
 
 	return initialiseLogging(cmd, &opts.LoggingOpts)
 }
 
+func initialiseOnlySpec(cmd *cli.Command, opts *grizzly.Opts) *cli.Command {
+	cmd.Flags().BoolVarP(&opts.OnlySpec, "only-spec", "s", false, "this flag is only used for dashboards to output the spec")
+	cmdRun := cmd.Run
+	cmd.Run = func(cmd *cli.Command, args []string) error {
+		opts.HasOnlySpec = cmd.Flags().Changed("only-spec")
+		return cmdRun(cmd, args)
+	}
+
+	return cmd
+}
+
```

Comment:
Because we want to know whether the flag is set or not. A bool only has true/false by default which means we can't tell from that whether we should defer to context. This adds a little code, executed *after* parsing has been done, that checks whether there was or wasn't an `only-spec` flag set.